File: VB/ExportOnlyBodyContent/Form1.vb
Imports System
Imports System.IO
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
'#Region "#usings"
Imports DevExpress.XtraRichEdit.Export
Imports DevExpress.XtraRichEdit.Export.Html
Imports DevExpress.Office.Services
Imports System.Runtime.InteropServices

'#End Region  ' #usings
Namespace ExportOnlyBodyContent

    Public Partial Class Form1
        Inherits XtraForm

        Private cssExportType As CssPropertiesExportType

        Private htmlExportType As ExportRootTag

        Public Sub New()
            InitializeComponent()
            InitComboHtmlExportType()
            InitComboCssExportType()
        End Sub

'#Region "Initializing"
        Private Sub InitComboHtmlExportType()
            Dim collExportHtml As ComboBoxItemCollection = edtExportHtmlType.Properties.Items
            collExportHtml.BeginUpdate()
            collExportHtml.Clear()
            collExportHtml.Add(ExportRootTag.Body)
            collExportHtml.Add(ExportRootTag.Html)
            collExportHtml.EndUpdate()
            edtExportHtmlType.SelectedIndex = 0
            htmlExportType = ExportRootTag.Body
        End Sub

        Private Sub InitComboCssExportType()
            Dim collCssStyle As ComboBoxItemCollection = edtCssStyleType.Properties.Items
            collCssStyle.BeginUpdate()
            collCssStyle.Clear()
            collCssStyle.Add(CssPropertiesExportType.Style)
            collCssStyle.Add(CssPropertiesExportType.Link)
            collCssStyle.Add(CssPropertiesExportType.Inline)
            collCssStyle.EndUpdate()
            edtCssStyleType.SelectedIndex = 0
            cssExportType = CssPropertiesExportType.Style
        End Sub

        Protected Friend Sub OnLoadDocumentClick(ByVal sender As Object, ByVal e As EventArgs)
            richEditControl1.LoadDocument()
        End Sub

        Private Sub richEditControl_DocumentLoaded(ByVal sender As Object, ByVal e As EventArgs)
            Try
                Dim fileName As String = richEditControl1.Options.DocumentSaveOptions.CurrentFileName
                If Not String.IsNullOrEmpty(fileName) Then
                    Using reader As StreamReader = New StreamReader(fileName)
                        memoEdit1.Text = reader.ReadToEnd()
                    End Using
                End If
            Catch
            End Try
        End Sub

        Private Sub richEditControl_EmptyDocumentCreated(ByVal sender As Object, ByVal e As EventArgs)
            memoEdit1.Text = String.Empty
        End Sub

'#End Region  ' Initializing
'#Region "Adjusting"
        Private Function GetFileName(ByVal filter As String) As String
            Using saveFileDialog As SaveFileDialog = New SaveFileDialog()
                saveFileDialog.Filter = filter
                saveFileDialog.RestoreDirectory = True
                saveFileDialog.CheckFileExists = False
                saveFileDialog.CheckPathExists = True
                saveFileDialog.OverwritePrompt = True
                saveFileDialog.DereferenceLinks = True
                saveFileDialog.ValidateNames = True
                If saveFileDialog.ShowDialog(Me) = DialogResult.OK Then Return saveFileDialog.FileName
            End Using

            Return String.Empty
        End Function

        Private Sub SaveFile(ByVal fileName As String, ByVal value As String)
            Using stream As FileStream = New FileStream(fileName, FileMode.Create, FileAccess.Write)
                Using writer As StreamWriter = New StreamWriter(stream)
                    writer.Write(value)
                End Using
            End Using
        End Sub

        Private Sub edtExportHtmlType_SelectedValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            htmlExportType = CType(edtExportHtmlType.EditValue, ExportRootTag)
        End Sub

        Private Sub edtCssStyleType_SelectedValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            cssExportType = CType(edtCssStyleType.EditValue, CssPropertiesExportType)
        End Sub

'#End Region
        Private Sub OnExportHtmlClick(ByVal sender As Object, ByVal e As EventArgs)
            Dim fileName As String = GetFileName("HyperText Markup Language Format|*.html")
            If String.IsNullOrEmpty(fileName) Then Return
            Dim svc As IUriProviderService = CType(richEditControl1.GetService(GetType(IUriProviderService)), IUriProviderService)
            svc.RegisterProvider(New MyUriProvider(Path.GetDirectoryName(fileName)))
            Dim stringHtml As String = String.Empty
            ExportHtml(stringHtml, Nothing, fileName)
            memoEdit2.Text = stringHtml
            SaveFile(fileName, stringHtml)
        End Sub

'#Region "#exporting"
        Private Sub ExportHtml(<Out> ByRef stringHtml As String, ByVal exporter As HtmlExporter, ByVal fileName As String)
            stringHtml = String.Empty
            Dim options As HtmlDocumentExporterOptions = New HtmlDocumentExporterOptions()
            options.ExportRootTag = htmlExportType
            options.CssPropertiesExportType = cssExportType
            options.TargetUri = Path.GetFileNameWithoutExtension(fileName)
            exporter = New HtmlExporter(richEditControl1.Model, options)
            stringHtml = exporter.Export()
        End Sub
'#End Region  ' #exporting
    End Class
End Namespace

File: CS/Form1.cs
using System;
using System.IO;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
#region #usings
using DevExpress.XtraRichEdit.Export;
using DevExpress.XtraRichEdit.Export.Html;
using DevExpress.XtraRichEdit.Commands;
using DevExpress.Office.Services;
#endregion #usings



namespace ExportOnlyBodyContent {
    public partial class Form1 : XtraForm {
        CssPropertiesExportType cssExportType;
        ExportRootTag htmlExportType;

        public Form1() {
            InitializeComponent();
            InitComboHtmlExportType();
            InitComboCssExportType();

        }
        #region Initializing
        private void InitComboHtmlExportType() {
            ComboBoxItemCollection collExportHtml = edtExportHtmlType.Properties.Items;
            collExportHtml.BeginUpdate();
            collExportHtml.Clear();
            collExportHtml.Add(ExportRootTag.Body);
            collExportHtml.Add(ExportRootTag.Html);
            collExportHtml.EndUpdate();
            edtExportHtmlType.SelectedIndex = 0;
            htmlExportType = ExportRootTag.Body;
        }
        private void InitComboCssExportType() {
            ComboBoxItemCollection collCssStyle = edtCssStyleType.Properties.Items;
            collCssStyle.BeginUpdate();
            collCssStyle.Clear();
            collCssStyle.Add(CssPropertiesExportType.Style);
            collCssStyle.Add(CssPropertiesExportType.Link);
            collCssStyle.Add(CssPropertiesExportType.Inline);
            collCssStyle.EndUpdate();
            edtCssStyleType.SelectedIndex = 0;
            cssExportType = CssPropertiesExportType.Style;
        }
        protected internal void OnLoadDocumentClick(object sender, EventArgs e) {
            this.richEditControl1.LoadDocument();
        }
        private void richEditControl_DocumentLoaded(object sender, EventArgs e) {
            try {
                string fileName = richEditControl1.Options.DocumentSaveOptions.CurrentFileName;
                if(!String.IsNullOrEmpty(fileName)) {
                    using(StreamReader reader = new StreamReader(fileName)) {
                        this.memoEdit1.Text = reader.ReadToEnd();
                    }
                }
            }
            catch {
            }
        }

        private void richEditControl_EmptyDocumentCreated(object sender, EventArgs e) {
            this.memoEdit1.Text = String.Empty;
        }
#endregion Initializing

        #region Adjusting
        private string GetFileName(string filter) {
            using(SaveFileDialog saveFileDialog = new SaveFileDialog()) {
                saveFileDialog.Filter = filter;
                saveFileDialog.RestoreDirectory = true;
                saveFileDialog.CheckFileExists = false;
                saveFileDialog.CheckPathExists = true;
                saveFileDialog.OverwritePrompt = true;
                saveFileDialog.DereferenceLinks = true;
                saveFileDialog.ValidateNames = true;
                if(saveFileDialog.ShowDialog(this) == DialogResult.OK)
                    return saveFileDialog.FileName;
            }
            return String.Empty;
        }
        private void SaveFile(string fileName, string value) {
            using(FileStream stream = new FileStream(fileName, FileMode.Create, FileAccess.Write)) {
                using(StreamWriter writer = new StreamWriter(stream)) {
                    writer.Write(value);
                }
            }
        }

        private void edtExportHtmlType_SelectedValueChanged(object sender, EventArgs e) {
            this.htmlExportType = (ExportRootTag)edtExportHtmlType.EditValue;

        }
        private void edtCssStyleType_SelectedValueChanged(object sender, EventArgs e) {
            this.cssExportType = (CssPropertiesExportType)edtCssStyleType.EditValue;
        }
        #endregion

        private void OnExportHtmlClick(object sender, EventArgs e) {
            string fileName = GetFileName("HyperText Markup Language Format|*.html");
            if(String.IsNullOrEmpty(fileName))
                return;
            IUriProviderService svc = (IUriProviderService) richEditControl1.GetService(typeof(IUriProviderService));
            svc.RegisterProvider(new MyUriProvider(Path.GetDirectoryName(fileName)));
            string stringHtml = String.Empty;
            ExportHtml(out stringHtml, null, fileName);
            this.memoEdit2.Text = stringHtml;
            SaveFile(fileName, stringHtml);

        }
        #region #exporting
    private void ExportHtml(out string stringHtml, HtmlExporter exporter, string fileName) {
        stringHtml = String.Empty;
        HtmlDocumentExporterOptions options = new HtmlDocumentExporterOptions();
        options.ExportRootTag = htmlExportType;
        options.CssPropertiesExportType = cssExportType;
        options.TargetUri = Path.GetFileNameWithoutExtension(fileName);
        exporter = new HtmlExporter(this.richEditControl1.Model, options);
        stringHtml = exporter.Export();
    }
        #endregion #exporting
    }
}